assignments/object-functions/code.js

```diff
@@ -0,0 +1,41 @@
+const getX = (object) => { return object.x }
+const point = (x, y) => {
+  let point = { 'x': x, 'y': y }
+  return point
+}
+const emptyObject = () => {
+  let emptyObject = {}
+  return emptyObject
+}
+const distance = (point1, point2) => {
+  return Math.sqrt((point2.x - point1.x) ** 2 + (point2.y - point1.y) ** 2)
+}
+const midpoint = (point1, point2) => {
+  return { 'x': (point1.x + point2.x) / 2, 'y': (point1.y + point2.y) / 2 }
+}
+const sumSalaries = (x) => {
+  let sum = 0
+  for (let i = 0; i < x.length; i++) {
+    sum = sum + x[i]['salary']
+  }
+  return sum
+}
+const newHighScore = (current, scores) => {
+  let max = current
+  for (let i = 0; i < scores.length; i++) {
+    if (scores[i]['score'] > max) {
+      max = scores[i]['score'];
+    }
+  }
+  return max
+}
+const summarizeBooks = (books) => {
+  let titles = []
+  let pages = 0
+  for (let i = 0; i < books.length; i++) {
+    titles[i] = books[i]['title']
+    pages = pages + books[i]['pages']
+  }
+  let summary = { 'titles': titles, 'pages': pages }
+  return summary
+}
```

Comment:
Just to point out, in case it's not obvious, you could also create the summary object at the beginning and then update it in the loop.

```javascript
const summarizeBooks = (books) => {
  const summary = { titles: [], pages: 0 }
  for (let i = 0; i < books.length; i++) {
    summary.titles[i] = books[i]['title']
    summary.pages = summary.pages + books[i]['pages']
  }
  return summary
}
```

Also, this line works in Javascript:

```javascript
    titles[i] = books[i]['title']
```

because Javascript is kind of loose about letting you assign to array elements that don't exist yet (i.e. `i` is not a valid index into `titles`). Other languages (like Java) are more picky. Safer and a bit more expressive is to write it like this:

```javascript
    titles.push(books[i]['title']);
```

This makes it clear that you're adding the title to the array, not setting an existing element in the array.

Finally, this is not something you probably know yet, but because it's so common to want to add a value to a variable (i.e. set the variable to its current value plus some amount), there's a shorthand for it:

Instead of:

```javascript
    pages = pages + books[i]['pages']
```

you can write:

```javascript
    pages += books[i]['pages']
```

Put it all together and switch to dot-property notation and you get this:


```javascript
const summarizeBooks = (books) => {
  const summary = { titles: [], pages: 0 }
  for (let i = 0; i < books.length; i++) {
    summary.titles.push(books[i].title)
    summary.pages += books[i].pages
  }
  return summary
}
```